packages/math-input/src/components/input/math-input.tsx

```diff
@@ -13,18 +13,21 @@ import {
     wonderBlocksBlue,
     offBlack,
 } from "../common-style";
+import ProvidedKeypad from "../provided-keypad";
 
 import CursorHandle from "./cursor-handle";
 import DragListener from "./drag-listener";
 import MathWrapper from "./math-wrapper";
 import {scrollIntoView} from "./scroll-into-view";
 
+import type {Cursor} from "../../types";
+
 const constrainingFrictionFactor = 0.8;
 
 type Props = {
-    keypadElement: any;
+    keypadElement: ProvidedKeypad;
     onBlur: () => void;
-    onChange: any;
+    onChange: (value: string, cb: any) => void;
```

Comment:
One... tiny... request.  Does it work if you define `cb` as:
```suggestion
    onChange: (value: string, callback: (any) => void) => void;
```
This'll help readers know it's a function and not something else.  Also, the naming is a bit obscure, so maybe `callback` instead?

Reply:
I'll switch the name, but the type is problematic. Anytime I touch it, errors start to spread. For instance somewhere in the same file they're setting it to a boolean.